pkg/ebpf/events_engine.go

```diff
@@ -0,0 +1,81 @@
+package ebpf
+
+import (
+	"context"
+
+	"github.com/aquasecurity/tracee/pkg/events"
+	"github.com/aquasecurity/tracee/pkg/rules/engine"
+	"github.com/aquasecurity/tracee/types/protocol"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+// engineEvents stage in the pipeline allows rules detection to be executed in the pipeline
+func (t *Tracee) engineEvents(ctx context.Context, in <-chan *trace.Event) (<-chan *trace.Event, <-chan error) {
+	out := make(chan *trace.Event)
+	errc := make(chan error, 1)
+	done := make(chan bool)
+	engineInput := make(chan protocol.Event)
+
+	engineOutput := engine.StartPipeline(t.config.EngineConfig, engineInput, done)
+
+	// TODO: in the upcoming releases, the rule engine should be changed to receive trace.Event,
+	// and return a trace.Event, which should remove the necessity of converting trace.Event to protocol.Event,
+	// and converting detect.Finding into trace.Event
+
+	go func() {
+		defer close(out)
+		defer close(errc)
+		defer close(engineInput)
+
+		for {
+			select {
+			case event := <-in:
+				id := events.ID(event.EventID)
+
+				// if the even is marked as submit, we pass it to the engine
+				if t.events[id].submit {
+					err := t.parseArguments(event)
+					if err != nil {
+						t.handleError(err)
+						continue
+					}
+
+					// pass the event to the sink stage, if the event is also marked as emit
+					// it will be sent to print by the sink stage
+					out <- event
+
+					// send the event to the rule event
+					engineInput <- event.ToProtocol()
```

Comment:
Check for the bug I found for https://github.com/aquasecurity/tracee/pull/2463. You have to call `t.shouldProcessEvent()` for the event, before sending it to the rules engine. Or else, if we are filtering through the cmdline, ALL the events will be passed to the rules engine (and not only those being filtered, which is what we want).